keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py

```diff
@@ -0,0 +1,223 @@
+            spatial_pyramid_pooling = SpatialPyramidPooling(
+                dilation_rates=[6, 12, 18]
+            )
+        spp_outputs = spatial_pyramid_pooling(backbone_outputs)
+
+        low_level_feature_extractor = keras.Model(
+            inputs=backbone.input,
+            outputs=backbone.get_layer(
+                backbone.pyramid_level_inputs["P2"]
+            ).output,
+            name="low-level-feature-extractor",
+        )
+        low_level_feature_projector = keras.Sequential(
+            [
+                keras.layers.Conv2D(
+                    name="low_level_feature_conv",
+                    filters=projection_filters,
+                    kernel_size=1,
+                    padding="same",
+                    use_bias=False,
+                ),
+                keras.layers.BatchNormalization(name="low_level_feature_norm"),
+                keras.layers.ReLU(name="low_level_feature_relu"),
+            ]
+        )
+
+        low_level_features = low_level_feature_extractor(inputs)
+        low_level_projected_features = low_level_feature_projector(
+            low_level_features
+        )
+
+        encoder_outputs = keras.layers.UpSampling2D(
+            size=(
+                low_level_projected_features.shape[1] // spp_outputs.shape[1],
+                low_level_projected_features.shape[2] // spp_outputs.shape[2],
+            ),
+            interpolation="bilinear",
+            name="encoder_output_upsampling",
+        )(spp_outputs)
+
+        combined_encoder_outputs = keras.layers.Concatenate(axis=-1)(
+            [encoder_outputs, low_level_projected_features]
+        )
+
+        if segmentation_head is None:
+            segmentation_head = keras.Sequential(
+                [
+                    keras.layers.Conv2D(
+                        name="segmentation_head_conv",
+                        filters=256,
+                        kernel_size=1,
+                        padding="same",
+                        use_bias=False,
+                    ),
+                    keras.layers.BatchNormalization(
+                        name="segmentation_head_norm"
+                    ),
+                    keras.layers.ReLU(name="segmentation_head_relu"),
+                    keras.layers.UpSampling2D(
+                        size=(4, 4), interpolation="bilinear"
+                    ),
+                ]
+            )
+
+            if dropout:
+                segmentation_head.add(
+                    keras.layers.Dropout(
+                        dropout, name="segmentation_head_dropout"
+                    )
+                )
+
+            # Classification layer
+            segmentation_head.add(
+                keras.layers.Conv2D(
+                    name="segmentation_output",
+                    filters=num_classes,
+                    kernel_size=1,
+                    use_bias=False,
+                    padding="same",
+                    activation="softmax",
```

Comment:
Whoa does this work!?!? Does `Conv2D` know to apply the softmax across the channels axis? I'm used to simple activations like `relu` so not familiar with this.

Reply:
Should I decouple the activate from `Conv2D`?

Reply:
Not if it works! Just curious 😄

Reply:
I tested, it seems to be working 😅
You can check some training logs [here](https://wandb.ai/geekyrakshit/deeplabv3-keras-cv/workspace).